syntax = "proto3";
import "Msg_Base.proto";

message GmStopServerReq {
    uint32    Data        	= 1;
}

message GmStopServerAck {
    uint32    RetCode       = 1;
}

message SvrRegToSvrReq {
    int32     ServerID      = 1;
    string    ServerName    = 2;
    string    ServerIp      = 3;
    int32     ServerPort    = 4;
    int32     HttpPort      = 5;
}

message SvrRegToSvrAck {
    uint32    RetCode       = 1;
    string    NewSvrName    = 2;
}

message RoleDisconnectReq {
    uint64    AccountID    = 1;
    uint64    RoleID       = 2;
}

message RoleDisconnectAck {
    uint32    RetCode = 1;
}

message RoleReconnectReq {
    uint64    RoleID    = 1;
    uint64    AccountID = 2;
    uint32    LoginCode = 3;
}

message RoleReconnectAck {
    uint32    RetCode = 1;
}

message RemoveConnectNty {
    uint64    RoleID = 1;
}

///创建场景
message CreateNewSceneReq{
    int32    CopyID        = 1;
    int32    CopyType      = 2;
    int32    PlayerNum     = 3;
    uint64   CreateParam   = 4;
}

///创建场景
message CreateNewSceneAck{
    uint32    RetCode        = 1;
    int32     CopyID         = 2;
    int32     CopyGuid       = 3;
    uint64    CreateParam    = 4;
    int32     ServerID       = 5;
    int32     CopyType       = 6;
    int32     PlayerNum      = 7;
}

////
message EnterSceneReq {
    uint64   RoleID      = 1;
    int32    ServerID    = 2;  //场景服ID
    int32    CopyGuid    = 3;  //副本实例ID
    int32    CopyID      = 4;  //副本类型ID
}

message EnterSceneAck {
    uint32   RetCode        = 1;
    int32    CopyID         = 2;
    int32    CopyGuid       = 3;
    //以下是主角创建需要信息
    uint64   RoleID         = 4;
    int32    Carrer         = 5;
    int32    ActorID        = 6;
    string   RoleName       = 7;
    int32    Camp           = 8;
    float    X              = 9;
    float    Y              = 10;
    float    Z              = 11;
    float    Ft             = 12;
    int32    Hp             = 13;
    int32    Mp             = 14;
    int32    HpMax          = 15;
    int32    MpMax          = 16;
    int32    Speed          = 17;
    int32    MountID        = 18;
    repeated int32 Equips   = 19; //武器
    repeated SkillItem Skills  = 20;//技能
}

message BackToCityReq {
    uint32    Param = 1;
}

message BackToCityAck {
    uint32    RetCode = 1;
}

message LeaveSceneReq {
    uint64   RoleID = 1;
    int32    Reason = 2;
}

message LeaveSceneAck {
    uint32    RetCode = 1;
}

message AbortSceneReq {
    uint64   RoleID = 1;
    int32    Reason = 2;
}

message AbortSceneAck {
    uint32    RetCode = 1;
}

message AbortSceneNty {
    uint64   RoleID      = 1;
    int32    CopyID      = 2;  //副本类型
    int32    CopyGuid    = 3;
    int32    CopyType    = 4;
    int32    ServerID    = 5;
    int32    Param       = 6;
}

message TransRoleData {
    uint64   RoleID         = 1;
    string   Name           = 2;
    int32    CarrerID       = 3;
    int32    ActorID        = 4;
    int32    Level          = 5;
    int32    MountID        = 6;
    int32    Robot          = 7; //是否是机器人
    repeated int32 Equips   = 8;
    repeated int32 Propertys= 9;
    repeated SkillItem Skills = 10;
}

message TransPetData {
    uint64   PetGuid            = 1;
    int32    PetID              = 2;
    int32    ActorID            = 3;
    string   Name               = 4;
    int32    Level              = 5;
    repeated int32 Propertys    = 6;
    repeated SkillItem  Skills  = 7;
}


message TransPartnerData {
    uint64   PartnerGuid        = 1;
    int32    PartnerID          = 2;
    int32    ActorID            = 3;
    string   Name               = 4;
    int32    Level              = 5;
    repeated int32 Propertys    = 6;
    repeated SkillItem  Skills  = 7;
}

message TransferDataItem {
    int32               Camp        = 1;            //战场中的阵营
    TransRoleData       RoleData    = 2;
    TransPetData        PetData     = 3;
    TransPartnerData    PartnerData = 4;
}

message TransferDataReq {
   repeated TransferDataItem TransDatas = 1;
}

message TransferDataAck {
    uint32   RetCode        = 1;
    uint64   RoleID         = 2;
    int32    CopyID         = 3;
    int32    CopyGuid       = 4;
    int32    ServerID       = 5;
} 

message NotifyIntoScene {
    uint64   RoleID        = 1;
    int32    CopyID        = 2;
    int32    CopyGuid      = 3;
    int32    ServerID      = 4;
}

message NearByRoleItem {
    uint64   RoleID    = 1;
    int32    ActorID   = 2;
    int32    Level     = 3;
    string   Name      = 4;
}

message CopyInsItem {
    uint32   CopyGuid    = 1;
    int32    CopyID      = 2;
    int32    CopyType    = 3;
    int32    ServerID    = 4;
}

message CopyReportReq 
{
    int32    ServerID = 1;
    repeated CopyInsItem CopyList = 2;
}

message UseHpBottleReq {

}

message UseHpBottleAck {
    uint32   RetCode = 1;
    int32    Count   = 2;
}

message UseMpBottleReq {
}

message UseMpBottleAck {
    uint32   RetCode = 1;
    int32    Count   = 2;
}

message ChatMessageReq {
    string    SrcName        = 1;
    uint64    SrcID          = 2;
    uint64    TargetID       = 3; //目标ID
    int32     ChatChl        = 4; //聊天频道; 
    uint64    GuildID        = 5; //公会ID
    string    Content        = 6;
}

message ChatMessageAck {
    uint32    RetCode = 1;
}

message ChatMessageNty {
    string    SrcName        = 1;
    uint64    SrcID          = 2;
    int32     SrcLevel       = 3;
    uint64    TargetID       = 4; //目标ID
    int32     ChatChl        = 5;
    uint64    GuildID        = 6;
    uint64    TimeStamp      = 7;
    string    Content        = 8;
}

message CommonReq {
    uint32    Param1    = 1;
}

message CommonAck {
    uint32    RetCode    = 1;
}

message BroadMessageNotify {
    uint32      MsgID            = 1;
    repeated    uint32 ConnID    = 2;
    bytes       MsgData          = 3;    
}

message LogicRegToLoginReq {
    int32    ServerID      = 1;
    string   ServerName    = 2;
    string   SvrInnerIp    = 3;
    int32    ServerPort    = 4;
    int32    HttpPort      = 5;
}

message LogicRegToLoginAck {
    uint32    RetCode       = 1;
    string    NewSvrName    = 2;
    uint64    SvrOpenTime   = 3; //开服时间
}

message LogicRegToDbSvrReq {
    int32    ServerID      = 1;
    string   ServerName    = 2;
    uint32   ProcessID     = 3;
}

message LogicRegToDbSvrAck {
    uint32    RetCode       = 1;
}

message LogicUpdateInfoReq {
    int32     ServerID     = 1;
    string    ServerName   = 2;
    int32     MaxOnline    = 3;
    int32     CurOnline    = 4;
    int32     TotalNum     = 5;
    int32     CacheNum     = 6;
    int32     Status       = 7;
    int32     DbErrCnt     = 8; //数据库写失败次数
}

message LogicUpdateInfoAck {
    uint32    RetCode       = 1;
    uint64    SvrOpenTime   = 2; //开服时间
}

message SetupEquipReq {
    uint64    EquipGuid = 1;
    uint64    BagGuid = 2;
}

message SetupEquipAck {
    uint32    RetCode    = 1;
}

message Msg_GiftCodeReq{
    uint64  RoleID     = 1;
    string  GiftCode   = 2;
    uint32  Channel    = 3;
    uint64  AccountID  = 4;
    int32   AreaID     = 5;
    
}

message Msg_GiftCodeAck{
    uint32   RetCode = 1;
    uint64   RoleID  = 2;
    repeated ItemData items = 3;
}

message UnsetEquipReq {
    uint64    EquipGuid = 1;
}

message UnsetEquipAck {
    uint32    RetCode    = 1;
}

message SetupGemReq {
    uint64    GemGuid        = 1;
    uint64    BagGuid        = 2;
    int32     TargetPos    = 3;
}

message SetupGemAck {
    uint32    RetCode    = 1;
}

message UnsetGemReq {
    uint64    GemGuid = 1;
}

message UnsetGemAck {
    uint32    RetCode    = 1;
}

message SetupPartnerReq {
    uint64    PartnerGuid    = 1;
    int32    TargetPos    = 2;
}

message SetupPartnerAck {
    uint32    RetCode    = 1;
}

message UnsetPartnerReq {
    int32    TargetPos = 1;
}

message UnsetPartnerAck {
    uint32    RetCode    = 1;
}

message SetupMountReq {
    uint64    MountGuid    = 1;
    int32     TargetPos    = 2;
}

message SetupMountAck {
    uint32    RetCode    = 1;
}

message UnsetMountReq {
    uint64    MountGuid = 1;
}

message UnsetMountAck {
    uint32    RetCode    = 1;
}


message SetupPetReq {
    uint64    PetGuid    = 1;
}

message SetupPetAck {
    uint32    RetCode    = 1;
}

message UnsetPetReq {
    uint64    PetGuid = 1;
}

message UnsetPetAck {
    uint32    RetCode    = 1;
}

message StoreBuyReq {
    int32    StoreType = 1;
    int32    StoreID = 2;
    int32    BuyNum = 3;
}

message StoreBuyAck {
    uint32    RetCode    = 1;
}

//玩家一些数据发生变化，由逻辑服通知战斗服
message ObjectChangeNotify {
    int64   RoleID      = 1;        //变化的角色ID
    int32   ChangeType  = 2; //变化的类型
    int64   IntValue1   = 3;
    int64   IntValue2   = 4;
    string  StrValue    = 5;
}

message HitEffectItem
{
    uint64            CastGuid    = 1; //施放者GUID
    uint64            TargetGuid  = 2; //目标GUID
    int32             HurtValue   = 3; //伤害值
    bool              Crit        = 4; //是否暴击
    uint32            HitActionID = 5;
    uint32            HitEffectID = 6;
    float             HitDistance = 7;
}

message HitEffectNtf
{
    repeated HitEffectItem ItemList = 1;
}

message UseItemReq {
}

message UseItemAck {
}

message MsgGetRandomNameReq {
    int32 Sex = 1;
}

message MsgGetRandomNameAck {
    string Name = 1;
}

message Msg_DbErrorCountNty{
    uint32  ErrorCount = 1;
}

message Msg_GameParamReq
{
    int32  Channel       =   1;      //渠道
    string ClientVer    =   2;      //版本
}

message Msg_GameParamAck
{
    repeated string ParamKey   = 1;      //参数key
    repeated string ParamValue = 2;     //参数值
}
